Code/Framework/AzCore/AzCore/Memory/SystemAllocator.cpp

```diff
@@ -234,4 +235,10 @@ namespace AZ
         #endif
     }
 
+    void SystemAllocator::GarbageCollect()
+    {
+        #if (AZCORE_SYSTEM_ALLOCATOR != AZCORE_SYSTEM_ALLOCATOR_MALLOC)
+            m_subAllocator->GarbageCollect();
+        #endif
+    }
```

Comment:
Could this not instead be `If (m_subAllocator) { subAllocator->GarbageCollect(); }` so that it works with future sub allocators or alternative ones?    Instead of using an ifdef, I mean.